app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Tab.kt

```diff
@@ -0,0 +1,443 @@
+import com.hedvig.android.design.system.hedvig.tokens.ColorSchemeKeyTokens.TextPrimary
+import com.hedvig.android.design.system.hedvig.tokens.LargeTabTokens
+import com.hedvig.android.design.system.hedvig.tokens.MediumTabTokens
+import com.hedvig.android.design.system.hedvig.tokens.MiniTabTokens
+import com.hedvig.android.design.system.hedvig.tokens.SmallTabTokens
+import kotlin.math.ceil
+
+// only for up to 6 TabItems!
+@Composable
+fun HedvigTabRowMaxSixTabs(
+  tabTitles: List<String>,
+  selectedTabIndex: Int,
+  onTabChosen: (index: Int) -> Unit,
+  modifier: Modifier = Modifier,
+  tabSize: TabSize = TabDefaults.defaultSize,
+  tabStyle: TabStyle = TabDefaults.defaultStyle,
+) {
+  var currentIndicatorOffset by remember { mutableStateOf(IntOffset(0, 0)) }
+  val indicatorOffset: IntOffset by animateIntOffsetAsState(
+    targetValue = currentIndicatorOffset,
+    animationSpec = tween(
+      durationMillis = 600,
+      easing = FastOutSlowInEasing,
+    ),
+    label = "",
+  )
+  var oneLineHeight by remember { mutableIntStateOf(0) }
+  Box(
+    modifier = modifier
+      .clip(tabSize.rowShape)
+      .background(tabStyle.colors.tabRowBackground)
+      .height(intrinsicSize = IntrinsicSize.Min)
+      .padding(tabSize.rowPadding),
+  ) {
+    Box(Modifier.withoutPlacement()) {
+      // to get one line height
+      HedvigText(
+        text = "measurement",
+        style = tabSize.textStyle,
+        modifier = Modifier
+          .onSizeChanged {
+            oneLineHeight = it.height
+          }
+          .padding(tabSize.tabPadding),
+      )
+    }
+    Layout(
+      contents = listOf(
+        {
+          TabIndicator(
+            indicatorColor = tabStyle.colors.chosenTabBackground,
+            indicatorShape = tabSize.tabShape,
+          )
+        },
+        {
+          tabTitles.forEachIndexed { index, title ->
+            TabItem(
+              modifier = Modifier
+                .clip(tabSize.tabShape),
+              onClick = {
+                onTabChosen(index)
+              },
+              text = title,
+              textStyle = tabSize.textStyle,
+              tabTextColor = tabStyle.colors.textColor,
+              contentPadding = tabSize.tabPadding,
+            )
+          }
+        },
+      ),
+    ) { allMeasurables: List<List<Measurable>>, constraints ->
+      val tabItemsMeasurables = allMeasurables[1]
+      if (tabItemsMeasurables.size <= 1) {
+        layout(0, 0) {}
+      } else {
+        // calculate the size
+        val fullWidth = constraints.maxWidth
+        val desiredItemWidth =
+          if (tabItemsMeasurables.any { it.minIntrinsicWidth(oneLineHeight) > fullWidth / 3 }) {
+            fullWidth / 2
+            // this is the maximum width we give the item, if the text is too big for the basic 1/3 of the full width.
+            // If it's still not enough, the text itself gets eclipsed later on
+          } else if (tabItemsMeasurables.size == 2) {
```

Comment:
Since we "short-circuit" into just splitting it half-half if we got specifically 2 items here, could we move the second check as the first check in this if/else chain?
It would be kinda like getting that "good scenario" out of the way first, and only after that is not true we iterate over the measurables to check their intrinsic width etc.